src/backend/Modules/Account/Timebox.Account.Application/Services/PBKDF2PasswordService.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Security.Cryptography;
+using System.Text.RegularExpressions;
+using Timebox.Account.Application.Interfaces.Services;
+
+namespace Timebox.Account.Application.Services
+{
+    public class PBKDF2PasswordService : IPasswordService
+    {
+        private const int _hashLength = 32;
+        private const int _base64HashLength = ((_hashLength * 4) / 3) + (_hashLength % 3);
+        private const int _saltLength = 32;
+        private const int _base64IterationsLength = 8;
+        
+        public bool IsStrongPassword(string password)
+        {
+            return Regex.IsMatch(password, @"^(?=.*[A-Z])(?=.*[_!@#$&*])(?=.*[0-9].*[0-9])(?=.*[a-z].*[a-z]).{8,}$");
+        }
+
+        public string HashPassword(string password, string salt = null, int? numberOfIterations = null)
+        {
+            salt ??= GenerateSalt();
```

Comment:
Oo what does this do ?

Reply:
If the salt is null it will set the value to whatever GenerateSalt is. This allows it to be unit tested.